modules/ldap_query/lambda/lambda.py

```diff
@@ -0,0 +1,467 @@
+                days = (today - pwd_last_set).days
+                user = {
+                    "name": user_obj['cn'][0],
+                    "email": user_obj['mail'][0],
+                    "dn": user_obj['distinguishedName'][0],
+                    "days_since_last_pwd_change": days
+                }
+                log.debug(f'got user: {user}')
+                # if employeeType is set to DTU assume the user is a test user
+                if days >= 120 or desc == "Test account":
+                    stale_users["120"].append(user)
+                elif days >= 90:
+                    stale_users["90"].append(user)
+                elif days >= 60:
+                    stale_users["60"].append(user)
+            except KeyError:
+                continue
+        # log.debug(f"retrieved the following stale users: {stale_users}")
+        return stale_users
+
+    def get_ldif(self):
+        """Creates a ldif document with the query results"""
+        # could be an alternative way of user disablement
+
+
+def get_file_name(file_name, extension):
+    timestamp = datetime.now().strftime("%Y_%m_%d_T%H%M%S.%f")
+    return f"{file_name}_{timestamp}.{extension}"
+
+
+def create_json_doc(content):
+    """create a table"""
+    # This can be fleshed out to make the retrieved information
+    # more user friendly if desired/required
+    artifact = {}
+    artifact['content'] = json.dumps(content)
+    artifact['file_name'] = get_file_name("user_expiration_table", "json")
+    artifact['raw_scan_results'] = True
+    return artifact
+
+
+def get_html_table_headers(user_list):
+    table_headers = []
+    try:
+        for key in user_list['120'][0].keys():
+            table_headers.append(key)
+        return table_headers
+    except IndexError:
+        # return the empty list of there are no accounts to be disabled
+        return table_headers
+
+
+def render_template(template="html_table.html", **kwargs):
+    env = Environment(
+        loader=FileSystemLoader(os.path.join(
+                os.path.dirname(__file__), 'templates'
+                ), encoding='utf8'))
+    template = env.get_template(template)
+    output_from_parsed_template = template.render(**kwargs)
+    return output_from_parsed_template
+
+
+def create_html_table(content):
+    # un-group the users for the html table
+    template_contents = {
+        "table_headers": get_html_table_headers(content),
+        "user_list": content
+    }
+    artifact = {}
+    artifact['content'] = render_template(**template_contents)
+    artifact['file_name'] = get_file_name("user_expiration", "html")
+    return artifact
+
+
+def generate_artifacts(content):
+    """Returns the list of objects to upload to s3"""
+    artifacts = {}
+    artifacts['user_expiration_table'] = create_json_doc(content)
+    artifacts['user_expiration_html'] = create_html_table(content)
+    return artifacts
+
+
+def put_object(dest_bucket_name, dest_object_name, src_data):
```

Comment:
This seems over-complicated. Put the object, let the function raise an exception. Use try/except in the caller to handle any error, rather than masking it by returning true/false

Reply:
agreed, updated